[oneBoz] iP

oneBoz opened the pull request:
# Friday
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Friday frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/oneBoz/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:
- [X] Add todo tasks without deadline
- [x] Add deadline tasks with deadline 
- [x] Add event tasks with duration
- [x] Change task completion status
- [x] Find and delete tasks

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Friday {
    public static void main(String[] args) {
        start();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Add Friday.java

Commit:
Added echo feature on Friday.java

Commit:
Added level 2 features

Commit:
Added level 2 features

Commit:
Added level 3 features

Commit:
Added level 4 features

Commit:
Updated test script

Commit:
Added level 5 features

Commit:
Added Level-6 features

Commit:
Added enums

Commit:
Add storage

Commit:
Add date and time

Commit:
Change Event toString

Commit:
Add Level-8 features

Commit:
Merge branch 'branch-Level-8'

Commit:
Add more OOP

Commit:
Add packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Automate project builds using Gradle

Gradle added into project

This allows for more convenience in the project building process

Let's use gradle together!

Gradle is used because of course instructions

Commit:
Add JUnit tests to test the behaviour of the code

Code does not have developer testing

Developer testing is important to reduce error, and troubleshooting
efforts

JUnit tests added to Storage.java and Parser.java

Let's make Friday bug free

Commit:
Fix bugs and changed String to create Jar file

Code uses text blocks

IntelliJ does not support text blocks when building Jar files

Let's change the String formats and replace all text blocks in order to
build a jar file.

Commit:
Add JavaDoc comments to methods and constructors

Added JavaDoc comments to the following methods:
  - createTaskFromInput
  - createCommandFromInput
  - getDesc
  - scan (overloaded versions)
  - write
  - read
  - Storage constructor
  - toString (Task | Event | Deadline)
  - toRawString (Task | Event | Deadline)
  - Task constructor
  - Deadline constructor
  - Event constructor
Enhanced documentation for better code understanding and maintenance

Commit:
Add JavaDoc comments to Task constructor

Added JavaDoc comments to Task constructor

Enhanced documentation and prevents misunderstanding that Task is a
concrete class

Commit:
Update code to adhere to coding standards

- Refactored code for improved readability and consistency
- Applied coding conventions for method and variable naming
- Ensured consistent use of formatting and indentation

Commit:
Add find feature to locate tasks

- Implemented task searching functionality based on task description
- Updated relevant classes and methods to support the find feature

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Add JavaDoc comment to contains method

- Documented the contains method to explain its functionality
- Added details on parameters and return value
- Improved code clarity and maintainability through documentation

iamdiluxedbutcooler commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,124 @@
+                        out.add(t);
+                        break;
+                    case 'D':
+                        String[] dtmp = line.split(" \\(by: ");
```
Should rename the temporary(?) variable `dtmp` to something more descriptive.

iamdiluxedbutcooler commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,124 @@
+                        out.add(d);
+                        break;
+                    case 'E':
+                        String[] etmp = line.split(" \\(from: ");
```
`etmp` should be something more descriptive

SeanWong2 replied:
Is there a better variable name you can use?

iamdiluxedbutcooler commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,124 @@
+                        break;
+                    case 'E':
+                        String[] etmp = line.split(" \\(from: ");
+                        String[] tmp2 = etmp[1].split(" to: ");
```
I think `temp` should sounds more descriptive than `tmp`, but after all using `temp` violates _Use name to explain_ rule. 

Also, why it is temp 2 btw?

iamdiluxedbutcooler commented on src/main/java/ui/Main.java:
```diff
@@ -0,0 +1,7 @@
+import java.util.Scanner;
+
+public class Main {
+    
```
What is this class doing?

iamdiluxedbutcooler commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,162 @@
+    public static ArrayList<String> scan(String filepath) {
+        try {
+            Scanner sc = new Scanner(new File(filepath));
+            ArrayList<String> out = new ArrayList<>();
```
`out` can be something more descriptive like `tasks` (because it is going to store tasks from the scanned file right?)

iamdiluxedbutcooler commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,162 @@
+     * @param input the input string containing the command and description
+     * @return the description part of the input string, with leading and trailing whitespace removed
+     */
+    public static String getDesc(String input) {
```
should `getDescription()` be more descriptive?

iamdiluxedbutcooler reviewed (commented):
Overall LGTM! Just minor suggestions. Keep up on your good work! 🤩

SeanWong2 commented on src/main/java/enumeration/CommandType.java:
```diff
@@ -0,0 +1,28 @@
+package enumeration;
+
+public enum CommandType{
```
I like the use of splitting your commands into different types

SeanWong2 commented on src/main/java/Friday.java:
```diff
@@ -0,0 +1,198 @@
+import java.util.ArrayList;
+
+
+import exception.InvalidIndexException;
```
Should the imports be in alphabetical order?

SeanWong2 commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,71 @@
+    protected LocalDate toInDateFormat;
+    protected boolean isInDateFormat = false;
+
+    /**
```
I like the detailed description of your methods

Commit:
Integrate Checkstyle plugin

- Added Checkstyle plugin to Gradle build configuration to ensure consistent code style across the project.
- Configured custom Checkstyle rules to align with project standards.
- Enabled automatic code style checks during the build process.

Commit:
Add JavaFX GUI for improved user interface

- Implemented JavaFX framework for the application's graphical user interface
- Designed initial layout with basic controls and interactions
- Integrated GUI with backend logic for enhanced user experience

SeanWong2 reviewed (commented):
Looks good in general

Commit:
Enhance code structure with more OOP principles

- Refactored code to improve encapsulation, inheritance, and abstraction
- Organized related functionalities into classes and objects
- Improved code readability and maintainability by adhering to OOP best practices

Commit:
Enhance code quality

- Improved code readability and structure
- Applied best practices for naming conventions and formatting
- Refactored methods for better maintainability and efficiency
- Added or updated Javadoc comments for better documentation

Commit:
Add assertions to validate functionality

- Added assertions to key methods to ensure correctness
- Improved test coverage by including edge cases and expected outcomes
- Enhanced reliability of tests with additional validation checks

oneBoz committed:
Merge pull request #1 from oneBoz/branch-A-Assertions

Add assertions to validate functionality

oneBoz committed:
Merge pull request #2 from oneBoz/branch-A-CodeQuality

Enhance code quality

Commit:
Add C-BetterSearch

Find command only return searchs if the query is exactly the same as
item

Find command should be more flexible and matches items even if keyword
matches the item only partially

Let's update the search algorithm to match partial keywords

oneBoz committed:
Merge pull request #3 from oneBoz/branch-BCD-Extension

Add C-BetterSearch

Commit:
Add User Guide for application usage

User guide is not updated

User guide needs to include all the features that Friday has

Adding an updated User Guide and a product screenshot would help new
users use the Friday chatbot properly

Let's
* update the User Guide and
* add a product screenshot

Commit:
Update User guide

User guide had typos

User guide needs to be as accurate as possible to avoid confusion

User guide will be read at least 3 times

Let's update the user guide

Commit:
Update User Guide

Some hyperlinks were incorrect

Incorrect hyperlinks may confuse users when they click into it

All the hyperlinks were checked and tested manually

Let's update the user guide

Commit:
Update README.md

Current README.md is using the template

Friday is now the chatbot and Duke is replaced

Change the README.md file using proper format

Let's update README.md file

Commit:
Update README file

Hyperlink in the current README file is outdated

It must lead to confusion for user

All hyperlinks are updated manually

Let's update README file

Commit:
Update Parser

Parser class does not check for invalid integer inputs for delete, mark
and unmark

Error message will not show for the user

Try catch blocks added to Parser to handle invalid inputs for delete,
mark, unmark

Let's update Parser class
